### PR TITLE
Ap 4671 flow refactor capital incomes

### PR DESCRIPTION
**Pull request by kmahern**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-4671)

Refactored the following steps:-

- client_completed_means
- employment_incomes
- unexpected_employment_incomes
- full_employment_details
- identify_types_of_income
- incoming_transactions
- regular_inomes
- cash_incomes
- student_finances
- income_summary

(I did not refactor a couple of the steps that were originally on the ticket. domestic_abuse_summaries had already been refactored, I think this might have been an error when the ticket was drafted. I moved housing_benefits to ap-4672 as it seemed to be a better fit with the outgoings tickets)

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- The standards in the [Git Workflow document on Confluence](https://dsdmoj.atlassian.net/wiki/spaces/ATPPB/pages/4602855954/Git+Workflow) should be followed
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: spec/services/flow/steps/provider_income/regular_incomes_step_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  describe "#path" do
+    subject { described_class.path.call(legal_aid_application) }
+
+    it { is_expected.to eql providers_legal_aid_application_means_regular_incomes_path(legal_aid_application) }
+  end
+
+  describe "#forward" do
+    subject(:forward_step) { described_class.forward.call(legal_aid_application) }
+
+    context "when applicant has income types other than housing benefit" do
```
Comment (colinbruce) [inline spec/services/flow/steps/provider_income/regular_incomes_step_spec.rb:15]: Should housing benefit be in this description? It's not clear from the setup or the controller tests that housing benefit is involved here

File: spec/services/flow/steps/provider_income/regular_incomes_step_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      before { allow(legal_aid_application).to receive(:income_types?).and_return(true) }
+
+      it { is_expected.to eq :cash_incomes }
+    end
+
+    context "when applicant does not have income types other than housing benefit" do
```
Comment (colinbruce) [inline spec/services/flow/steps/provider_income/regular_incomes_step_spec.rb:21]: ditto housing benefit

**Review by colinbruce — COMMENTED** [review]
Some questions on the test descriptions, hopefully nothing major

File: spec/services/flow/steps/provider_income/client_completed_means_step_spec.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    end
+
+    context "when HMRC returns an applicant_unexpected_employment_data status" do
+      let(:status) { :applicant_unexpected_employment_data }
+
+      it { is_expected.to eq :unexpected_employment_incomes }
+    end
+
+    context "when HMRC returns an applicant_not_employed status" do
+      before { allow(HMRC::StatusAnalyzer).to receive(:call).with(legal_aid_application).and_return(status) }
```
[the remark this answers is not in the document]
Comment (kmahern) [reply spec/services/flow/steps/provider_income/client_completed_means_step_spec.rb]: 👍 removed

File: spec/services/flow/steps/provider_income/regular_incomes_step_spec.rb (hunk 1 above)
Comment (kmahern) [reply spec/services/flow/steps/provider_income/regular_incomes_step_spec.rb]: 👍 was looking at income_types? on the legal_aid_application model which defines income_types? as transaction_types.credits.without_housing_benefits.any? but have removed.

File: spec/services/flow/steps/provider_income/regular_incomes_step_spec.rb (hunk 2 above)
Comment (kmahern) [reply spec/services/flow/steps/provider_income/regular_incomes_step_spec.rb]: 👍 was looking at income_types? on the legal_aid_application model which defines income_types? as transaction_types.credits.without_housing_benefits.any? but have removed.

**Review by colinbruce — APPROVED** [review]
LGTM 👍

**Pull request merged**